Only store `_edit_use_anchors_` metadata if value is not the default

Calinou opened the pull request:
`master` version of https://github.com/godotengine/godot/pull/58254.

The default value is assumed to be `false`, so this metadata only needs to be stored if the value is `true`.

This closes https://github.com/godotengine/godot/issues/57862.

*Bugsquad edit (keywords for easier searching): edit_use_anchors*

akien-mga commented on editor/plugins/canvas_item_editor_plugin.cpp:
```diff
@@ -4129,9 +4129,13 @@ void CanvasItemEditor::_set_anchors_and_offsets_to_keep_ratio() {
 			undo_redo->add_do_method(control, "set_anchor", SIDE_BOTTOM, bottom_right_anchor.y, false, true);
 			undo_redo->add_do_method(control, "set_meta", "_edit_use_anchors_", true);
```
What about this?

Calinou replied:
It always stores the value `true`, so I believe it should be left in.

rakkarage commented:
The pull request build seems to work for me. New scenes no longer have that setting written into them it seems.

But during testing I ran into another issue. You cannot undo a toggle of the anchor button? Maybe I will open a new issue?

Thanks.

Calinou commented:
> But during testing I ran into another issue. You cannot undo a toggle of the anchor button? Maybe I will open a new issue?

Does this happen without this PR?

rakkarage commented:
Yes, in 3 and 4.

- Make new Control scene and move control then toggle use anchors
- Scene->undo un-moves the control instead of the toggle

Calinou commented:
> Yes, in 3 and 4.
> 
>     * Make new Control scene and drag/move control then toggle use anchors
> 
>     * Scene->undo un-moves the control instead of the toggle

Use Anchors is an editor-only action, so I don't think it makes sense to be tracked in undo/redo.

rakkarage commented:
ya sorry I quess i still thought it was doing the the `_edit_use_anchors_` stuff :)

aaronfranke commented:
Really this should be moved out of the scene file. We don't need to store local editor specifics in version-controlled scene files, this kind of data would have a better home in the `.godot/editor/` folder.

pycbouh commented:
> Really this should be moved out of the scene file. We don't need to store local editor specifics in version-controlled scene files, this kind of data would have a better home in the `.godot/editor/` folder.

Do we have examples of other per-node properties stored in there? This is similar to `_edit_lock_`, which is also stored with nodes at the moment. There is also a question of how and if we should allow editing this programmatically, so that, say, a plugin could create a node and mark it to use anchors by default.

I agree in principle, but there is also an argument to be made that this needs to be synced and needs to be attached to the node.

pycbouh commented:
This PR needs a rebase/remake since the relevant code is now in `control_editor_plugin.cpp`. It should mostly be the same, IIRC, but I'd still appreciate an update before reviewing it.

Calinou committed:
Only store `_edit_use_anchors_` metadata if value is not the default

The default value is assumed to be `false`, so this metadata
only needs to be stored if the value is `true`.

YuriSizov reviewed (approved):
The change seems to be straightforward.

@rakkarage has a point though. In one case we save this editor-only flag as a part of UndoRedo, and in another we don't. But that's existing behavior, so I think it's a non-blocker for this PR.

akien-mga commented:
Thanks!

akien-mga commented:
Could use a dedicated backport for `3.x` as this code has changed somewhat.
